src/librustc/ty/subst.rs

```diff
@@ -298,20 +275,20 @@ impl<'a, 'gcx, 'tcx> Substs<'tcx> {
                        target_substs: &Substs<'tcx>)
                        -> &'tcx Substs<'tcx> {
         let defs = tcx.lookup_generics(source_ancestor);
-        Substs::new(tcx, target_substs.iter().chain(&self[defs.own_count()..]).cloned())
+        tcx.mk_substs(target_substs.iter().chain(&self[defs.own_count()..]).cloned())
```

Comment:
Can you clarify as to if it's possible to avoid `.cloned()` here? The `target_substs` and (I believe) the slice into `self` are both &Kind instead of Kind. Or is it Ty?